zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -511,43 +518,71 @@ def get_total_messages_and_usermessages(slack_data_dir: str, channel_name: str,
                 subtype = message['subtype']
                 if subtype in ["channel_join", "channel_leave", "channel_name"]:
                     continue
+                elif subtype == "file_share":
+                    total_attachments += 1
 
             for subscription in zerver_subscription:
                 if subscription['recipient'] == added_recipient[channel_name]:
                     total_usermessages += 1
             total_messages += 1
 
-    return total_messages, total_usermessages
+    return total_messages, total_usermessages, total_attachments
 
 def channel_message_to_zerver_message(constants: List[Any], channel: str,
                                       users: List[ZerverFieldsT], added_users: AddedUsersT,
                                       added_recipient: AddedRecipientsT,
                                       zerver_subscription: List[ZerverFieldsT],
                                       ids: List[int]) -> Tuple[List[ZerverFieldsT],
+                                                               List[ZerverFieldsT],
                                                                List[ZerverFieldsT]]:
     """
     Returns:
     1. zerver_message, which is a list of the messages
     2. zerver_usermessage, which is a list of the usermessages
     """
     slack_data_dir, REALM_ID = constants
-    message_id, usermessage_id = ids
+    message_id, usermessage_id, attachment_id = ids
     json_names = os.listdir(slack_data_dir + '/' + channel)
     zerver_message = []
     zerver_usermessage = []  # type: List[ZerverFieldsT]
+    zerver_attachment = []
 
     for json_name in json_names:
         messages = get_data_file(slack_data_dir + '/%s/%s' % (channel, json_name))
         for message in messages:
             has_attachment = False
+            url_private = None
             content, mentioned_users_id, has_link = convert_to_zulip_markdown(message['text'],
                                                                               users,
                                                                               added_users)
             rendered_content = None
+            user = get_message_sending_user(message)
             if 'subtype' in message.keys():
                 subtype = message['subtype']
                 if subtype in ["channel_join", "channel_leave", "channel_name"]:
                     continue
+                elif subtype == "file_share":
+                    has_attachment = True
+                    fileinfo = message['file']
+                    url_private = fileinfo['url_private']
+                    response = requests.get(url_private, stream=True)
+                    attachment_path = upload_message_image(
```

Comment:
Instead of the `upload_message_image`, the folder path can be calculated [folder path example](https://github.com/rheaparekh/slack2zulip-data-converter/tree/master/zulip-export-ws5m4j26/uploads/36) and uploaded file can be saved in this folder in the `output_dir` with the logs in [records.json](https://github.com/rheaparekh/slack2zulip-data-converter/blob/master/zulip-export-ws5m4j26/uploads/records.json). The import script will process this itself folder itself and create the attachment.

Reply:
I have a WIP PR for avatars https://github.com/zulip/zulip/pull/8349/files which is following the above method and correctly displays the avatar.